processing/src/main/java/io/druid/query/Query.java

```diff
@@ -99,4 +99,6 @@
   String getId();
 
   Query<T> withDataSource(DataSource dataSource);
+
+  Query<T> withDefaultTimeout(long defaultTimeout);
```

Comment:
The problem with #4131 was that it breaks compatibility with extension queries, doesn't it break it already? @gianm @cheddar